src/measure_theory/function/ae_eq_fun.lean

```diff
@@ -277,12 +277,62 @@ lift_rel_iff_coe_fn.symm
 instance [partial_order β] : partial_order (α →ₘ[μ] β) :=
 partial_order.lift to_germ to_germ_injective
 
-/- TODO: Prove `L⁰` space is a lattice if β is linear order.
-         What if β is only a lattice? -/
+section lattice
 
--- instance [linear_order β] : semilattice_sup (α →ₘ β) :=
--- { sup := comp₂ (⊔) (_),
---    .. ae_eq_fun.partial_order }
+variables [lattice β] [topological_space β] [topological_lattice β] [borel_space β]
+  [second_countable_topology β]
+
+instance : has_sup (α →ₘ[μ] β) :=
+{ sup := λ f g, ae_eq_fun.comp₂ (⊔) measurable_sup f g }
+
+instance : has_inf (α →ₘ[μ] β) :=
+{ inf := λ f g, ae_eq_fun.comp₂ (⊓) measurable_inf f g }
+
+lemma coe_fn_sup (f g : α →ₘ[μ] β) : ⇑(f ⊔ g) =ᵐ[μ] λ x, f x ⊔ g x := coe_fn_comp₂ _ _ _ _
+lemma coe_fn_inf (f g : α →ₘ[μ] β) : ⇑(f ⊓ g) =ᵐ[μ] λ x, f x ⊓ g x := coe_fn_comp₂ _ _ _ _
+
+private lemma le_sup_left (f g : α →ₘ[μ] β) : f ≤ f ⊔ g :=
+by { rw ← coe_fn_le, filter_upwards [coe_fn_sup f g], intros a ha,  rw ha, exact le_sup_left, }
+
+private lemma le_sup_right (f g : α →ₘ[μ] β) : g ≤ f ⊔ g :=
+by { rw ← coe_fn_le, filter_upwards [coe_fn_sup f g], intros a ha,  rw ha, exact le_sup_right, }
+
+private lemma sup_le (f g f' : α →ₘ[μ] β) (hf : f ≤ f') (hg : g ≤ f') : f ⊔ g ≤ f' :=
+begin
+  rw ← coe_fn_le at hf hg ⊢,
+  filter_upwards [hf, hg, coe_fn_sup f g],
+  intros a haf hag ha_sup,
+  rw ha_sup,
+  exact sup_le haf hag,
+end
+
+private lemma inf_le_left (f g : α →ₘ[μ] β) : f ⊓ g ≤ f :=
```

Comment:
If you add the `order_dual` instances I requested below, then proofs about `inf` can be replaced by `order_dual` proofs.

Reply:
I added a bunch or order_dual instances, but so far I haven't managed to write those proofs using `order_dual`. It looks like the issue is that `order_dual (α →ₘ[μ] β)` is not directly related to `α →ₘ[μ] (order_dual β)`.
I'd like help on that if you have time to look at it.

Reply:
I failed to use `order_dual` too. I changed TC assumptions to `has_measurable_sup₂` and `has_measurable_inf₂`.
Thanks!
bors d+